test/modules/auth2_0.py

```diff
@@ -33,21 +34,36 @@ def local_host_exception_setup():
         param = dumps({'username': 'admin', 'password': 'admin123', 'role': 'Administrator'})
         user_add_cmd = "curl -k -X POST -w '%{http_code}' -H 'Content-Type: application/json' -d '"
         user_add_cmd += param
-        user_add_cmd += "' https://127.0.0.1:" + USER_AUTH_PORT + "/api/2.0/users"
 
-        term = pxssh.pxssh()
-        term.SSH_OPTS = (term.SSH_OPTS
-                         + " -o 'StrictHostKeyChecking=no'"
-                         + " -o 'UserKnownHostsFile=/dev/null' ")
-        term.force_password = True
-        term.login(HOST_IP, SSH_USER, SSH_PASSWORD, port=SSH_PORT)
-        term.sendline(user_add_cmd)
-        index = term.expect(['201', '401', '403'], 10)
+        # add first user to remote rackhd directly
+        remote_user_add_cmd = user_add_cmd + \
+            "' https://{0}:{1}/api/2.0/users".format(HOST_IP, USER_AUTH_PORT)
+        try:
+            return_code = "unknown"
+            return_str = subprocess.check_output([remote_user_add_cmd], shell=True)
+            return_code = return_str[-3:]
+        except Exception as e:
+            LOG.info("Can't connect to RackHD https port directly, will set first user through ssh\n{0}".format(e))
+        if return_code in ['201', '401', '403']:
+            index = ['201', '401', '403'].index(return_code)
+        else:
+            # ssh login to rackhd and add first user to localhost rackhd
+            local_user_add_cmd = user_add_cmd + \
+                "' https://127.0.0.1:" + USER_AUTH_PORT + "/api/2.0/users"
```

Comment:
We really need to fix the 'USER_AUTH_PORT' - this is supposed to be the https port external view, but part of the CIT layout, so doesn't work when we use hardcoded 127.0.0.1 in this.  planning to fix this later on, fyi.